### PR TITLE
Filesystem operation common utility functions - IsExistingFile PathsInDir

### PR DESCRIPTION
**Pull request by dbenage-cx**

### CODE REVIEW

File: util/Directories.cpp [hunk 1]
```diff
@@ -635,3 +635,55 @@ fs::path GetPath(const std::string& path_string) {
     }
     return GetPath(path_type);
 }
+
+bool IsExistingFile(const fs::path& path) {
```
Comment (geoffthemedio) [inline util/Directories.cpp:639]: what does this do differently from just calling `fs::is_regular_file(path)` ?

**Comment by geoffthemedio** [comment]
Without being used in specific places, it's unclear why these are needed...

File: util/Directories.cpp (hunk 1 above)
Comment (dbenage-cx) [reply util/Directories.cpp:639]: The main use of such functions is work towards consolidating filesystem operations.
The are declaration compatible excerpts from larger work towards VFS.

Without considering the above (as it has not been discussed in depth), for this one function:
* Eliminates need to include boost/filesystem/operations.hpp at call site
* Short-circuits if `fs:exists` fails (with single stat call)
* Ensures filesystem errors are handled

**Comment by dbenage-cx** [comment]
`IsExistingPath` is picked from (EDIT, wrong pr) #1861
Both are used in [refactor of suitability report](https://github.com/freeorion/freeorion/compare/master...dbenage-cx:refactor_enc_suitability_report), which I hoped to rebase prior to posting as PR.

**Comment by dbenage-cx** [comment]
@geoffthemedio Prefer I rebase this branch (leaving just `PathsInDir`), or close this PR as invalid and keep the commit within other PR?

**Comment by geoffthemedio** [comment]
Within the other(s) seems preferable.

**Pull request closed without merging**